compiler/optimizer/OMRSimplifierHandlers.cpp

```diff
@@ -16534,25 +16715,123 @@ TR::Node *nullchkSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier *
 //
 TR::Node *divchkSimplifier(TR::Node * node, TR::Block * block, TR::Simplifier * s)
    {
-   // Remember the original child of the divchk. If simplification of this
-   // child causes it to be replaced by another node, the divchk is no longer
-   // needed.
-   // The child of the divchk may no longer be a divide node, due to commoning.
-   // In this case the divchk is no longer needed.
+   TR::Node * child = node->getFirstChild();
+   TR::Node * origChild = child;
+   TR::ILOpCode &childOpCode = child->getOpCode();
+
+   // Check whether the child has already been simplified
    //
-   TR::Node *originalChild = node->getFirstChild();
-   /////TR::ILOpCodes originalChildOpCode = originalChild->getOpCode().getOpCodeValue();
-   /////TR_ASSERT(originalChild->getVisitCount() != s->comp()->getVisitCount(),"Simplifier, bad divchk node");
-   TR::Node * child = originalChild;
-   if (originalChild->getVisitCount() != s->comp()->getVisitCount())
-      child = s->simplify(originalChild, block);
+   if (child->getVisitCount() != s->comp()->getVisitCount())
+      {
+      if (childOpCode.isDiv() || childOpCode.isRem())
+         {
+         s->_nodeToDivchk = NULL;
+
+         child = s->simplify(child, block);
 
-   if (child != originalChild ||
-       !(child->getOpCode().isDiv() || child->getOpCode().isRem()))
+         // Handshake with division and remainder simplifiers:  If simplifying the
+         // child of a DIVCHK results in a node that still needs to have a DIVCHK
+         // applied, those simplififiers will place that node in _nodeToDivchk.  If
+         // the simplification leaves no node that needs to have a DIVCHK applied,
+         // they will set _nodeToDivchk to NULL.  If _nodeToDivchk is non-null, that
+         // is used node as the child of the DIVCHK; otherwise, the DIVCHK is
```

Comment:
> that is used node as the child

Reply:
Fixed in commit 37c05e256709f96c1913b8fcab1b45b3e3b2dca2